### ISSUE TITLE
Make connect() wait for the stream to become writeable when using lazy connect and disabled offline queue

### ISSUE THREAD

**Issue by WRMSRwasTaken** [luin/ioredis#625]
Hi,

my ioredis configuration looks like this:

```js
this.main.redis = new Redis({
  host: nconf.get('redis:host'),
  port: nconf.get('redis:port'),
  password: nconf.get('redis:password'),
  db: nconf.get('redis:database'),
  keyPrefix: nconf.get('redis:prefix'),
  enableOfflineQueue: false,
  lazyConnect: true,
  retryStrategy() {
    return 5000;
  },
});
```

I am going to initialize the redis client in the following way to see if the redis connection succeeds on application startup :

```js
try {
  await this.main.redis.connect();
} catch (ex) {
  winston.error('Could not connect to the Redis backend!', ex);
  process.exit(1);
}

try {
  await this.main.redis.ping();
} catch (ex) {
  winston.error('Could not ping the Redis backend!', ex);
  process.exit(1);
}
```

Now the problem: connect() does return a promise which is resolved immediately (because it can connect to redis, the promise gets rejected if it can't), but it doesn't wait until the stream becomes writeable. So my code wil error on ping(), because the promise is getting rejected;

`
Could not ping the Redis backend! Error: Stream isn't writeable and enableOfflineQueue options is false`

In order to workaround this, I'd have to await a promise from `Bluebird` wrapping connect() and subscribing once to the `ready` event:

```js
await new Bluebird(async (resolve, reject) => {
  try {
    await this.main.redis.connect();
  } catch (ex) {
    reject(ex);
  }

  this.main.redis.once('ready', () => {
    resolve();
  });
});
```

**Comment by WRMSRwasTaken** [comment]
This is a bug in my opinion, or it should be clarified in the docs.

I'd suggest adding a listener to https://github.com/luin/ioredis/blob/master/lib/redis.js#L305 waiting for the `ready` event before resolving the promise, when those options are set like above.

**Comment by luin** [comment]
Hi, the behavior has been fixed in the beta release v4.0.0-0. Thank you for pointing this out!
